test/storage-path.test.js

```diff
@@ -0,0 +1,63 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright 2020 Joyent, Inc.
+ */
+
+var crypto = require('crypto');
+var uuidv4 = require('uuid/v4');
+var jsc = require('jsverify');
+
+var shark_client = require('../lib/shark_client.js');
+
+///--- Tests
+
+/**
+ * storagePathTest is a property test for the storagePath helper function in the
+ * shark_client module. jsverify generates 100 random strings, and checks that
+ * the property defined always holds. The property this tests is that the
+ * storagePath function properly forms a object storage path by breaking long
+ * object names into a series of subdirectories.
+ */
+exports.storagePathTest = function (t) {
+    function propStoragePath(objectNameLen) {
+        var objectName = 'a'.repeat(objectNameLen);
+        var objectNameHash =
+            crypto.createHash('md5').update(objectName).digest('hex');
+
+        var storagePathOpts = {
+            owner: uuidv4(),
+            bucketId: uuidv4(),
+            objectName: objectName,
+            objectNameHash: objectNameHash,
+            objectId: uuidv4(),
+            storageLayoutVersion: 2
+        };
+
+        var storagePath = shark_client.storagePath(storagePathOpts);
+
+        var actualPathParts = storagePath.substr(1).split('/');
+        var actualPathPartCount = actualPathParts.length;
+        var expectedObjectIdPrefix = storagePathOpts.objectId.substr(0, 2);
```

Comment:
nit: Could use the const for this `2`, or just add a comment with the const name if exporting it is a pain just for testing.

Reply:
Done